scripts/menu/client/cl_menu.lua

```diff
@@ -0,0 +1,554 @@
+        z = _z
+        break
+      end
+      Wait(0)
+    end
+  end
+  RequestCollisionAtCoord(x, y, z)
+  RequestAdditionalCollisionAtCoord(x, y, z)
+  SetPedCoordsKeepVehicle(ped, x, y, z)
+  DoScreenFadeIn(500)
+  SetGameplayCamRelativeHeading(0)
+end)
+
+-- [[ Teleport to the current waypoint ]]
+RegisterNetEvent('txAdmin:menu:tpToWaypoint', function()
+  local waypoint = GetFirstBlipInfoId(GetWaypointBlipEnumId())
+  if waypoint and waypoint > 0 then
+    local ped = PlayerPedId()
+    lastTp = GetEntityCoords(ped)
+    
+    DoScreenFadeOut(200)
+    while not IsScreenFadedOut() do Wait(1) end
+    
+    local blipCoords = GetBlipInfoIdCoord(waypoint)
+    debugPrint("waypoint blip: " .. json.encode(blipCoords))
+    local x = blipCoords[1]
+    local y = blipCoords[2]
+    local z = blipCoords[3]
+    for i = 0, 1000, 10 do
+      SetPedCoordsKeepVehicle(ped, x, y, i)
+      local zFound, _z = GetGroundZFor_3dCoord(x, y, i + 0.0)
+      if zFound then
+        debugPrint("3D ground found: " .. json.encode({ zFound, _z }))
+        z = _z
+        break
+      end
+      Wait(0)
+    end
+    RequestCollisionAtCoord(x, y, z)
+    RequestAdditionalCollisionAtCoord(x, y, z)
+    SetPedCoordsKeepVehicle(ped, x, y, z)
+    DoScreenFadeIn(500)
+    SetGameplayCamRelativeHeading(0)
+  else
+    sendSnackbarMessage("error", "You have no waypoint set!")
+  end
+end)
+
+--[[ Heal all players ]]
+RegisterNetEvent('txAdmin:menu:healed', function()
+  debugPrint('Received heal event, healing to full')
+  local ped = PlayerPedId()
+  local pos = GetEntityCoords(ped)
+  if IsEntityDead(ped) then
+    ResurrectPed(ped)
+    ped = PlayerPedId()
+    SetEntityCoords(ped, pos[1], pos[2], pos[3])
+  end
+  SetEntityHealth(ped, GetEntityMaxHealth(ped))
+end)
+
+--[[ Repair vehicle ]]
+RegisterNetEvent('txAdmin:menu:fixVehicle', function()
+  local ped = PlayerPedId()
+  local veh = GetVehiclePedIsIn(ped, false)
+  if veh and veh > 0 then
+    SetVehicleUndriveable(veh, false)
+    SetVehicleFixed(veh)
+    SetVehicleEngineOn(veh, true, false)
+    SetVehicleDirtLevel(veh, 0.0)
+    SetVehicleOnGroundProperly(veh)
+  end
+end)
+
+--[[ Spawn vehicles, with support for entity lockdown ]]
+RegisterNetEvent('txAdmin:menu:spawnVehicle', function(netID)
+  debugPrint(json.encode({ netID = netID }))
+  SetNetworkIdExistsOnAllMachines(netID, true)
+  SetNetworkIdCanMigrate(netID, true)
+  
+  -- get current veh and speed
+  local ped = PlayerPedId()
+  local oldVeh = GetVehiclePedIsIn(ped, false)
+  local oldVel = DoesEntityExist(oldVeh) and GetEntitySpeed(oldVeh) or 0.0
+  
+  -- only delete if the new vehicle is found
+  if oldVeh and IsPedInVehicle(ped, oldVeh, true) then
+    debugPrint("Deleting existing vehicle (" .. oldVeh .. ")")
+    SetEntityAsMissionEntity(oldVeh, true, true)
+    DeleteVehicle(oldVeh)
+  end
+  
+  local tries = 0
+  while NetworkGetEntityFromNetworkId(netID) == 0 do
```

Comment:
This'll log a warning if it doesn't exist - see NetworkDoesEntityExistWithNetworkId instead.